### PR TITLE
Add regex support functionality

### LINKED ISSUE #12923: [filebeat] Allow wildcards for drop_fields
**Describe the enhancement:**
`drop_fields.fields` should support glob or regex patterns. 

**Describe a specific use case for the enhancement or feature:**
I'm using the `add_docker_metadata` processor to extract container metadata, which also includes Docker labels. For some containers, there are quite a lot that don't contain relevant information for logging purposes. For example, the official `elasticsearch` image includes:

- "container.labels.license"
- "container.labels.org_label-schema_build-date"
- "container.labels.org_label-schema_license"
- "container.labels.org_label-schema_name"
- "container.labels.org_label-schema_schema-version"
- "container.labels.org_label-schema_url"
- "container.labels.org_label-schema_vcs-url"
- "container.labels.org_label-schema_vendor"
- "container.labels.org_label-schema_version"

Also many labels coming from `docker-compose` are not really needed and can be dropped as well. So you quickly end up with quite a list of labels, all starting with the same prefix.

(The exact same labels are also repeated as `docker.container.…`, but that's a different issue).

So instead of having to list all those labels under `drop_fields`, it would be nice to just write:

- "container.labels.license"
- "container.labels.org_label-schema*"

### PR DESCRIPTION
**Pull request by gonzaloe**
- Enhancement

## What does this PR do?

Added regex support for drop_fields processor as requested in #12923

Fields that are declared between slashes (as in Painless scripts) are going to be considered regex and all matches within field names will be removed

## Why is it important?

So deleting field-by-field with name can be avoided while using modules that declare a big amount of fields.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Configure a field using a regex pattern under *drop_fields* processor, on *fields* array, between slashes */*. For e.g.: 

```
drop_fields:
  fields: ["/field_to_be_deleted.*/"]
```

## Related issues

- Closes #12923

### CODE REVIEW

**Comment by rdner** [comment]
@kvch I suspect this PR will be affected by your refactoring here, right? https://github.com/elastic/beats/pull/31420

should we wait until the migration is merged?

**Comment by kvch** [comment]
Please wait. My PR is gigantic already.

**Comment by rdner** [comment]
@gonzaloe thank you for waiting, now the big change is merged, please update your PR.

**Comment by gonzaloe** [comment]
thanks for the notification @rdner, seems i'll have to move the mapstr implementation to elastic-agent-libs project, then import a new version here once that code is merged, is there a label to pause the review for now? or should i ping back once the other PR gets merged and everything working here again?

**Comment by gonzaloe** [comment]
@rdner in case you can take a quick look there -> https://github.com/elastic/elastic-agent-libs/pull/45

**Comment by rdner** [comment]
@gonzaloe it's approved, feel free to proceed.

**Comment by gonzaloe** [comment]
@rdner thanks! I've updated the dependency already :)
one more doubt so far:
- about the bug I pointed out on the drop_field package, should I go ahead and fix it, or is that code not being used? (deletion is being done while traversing the array from left to right)

**Comment by rdner** [comment]
@gonzaloe I have to ask you to run `goimport` on `libbeat/processors/actions/drop_fields.go` and `libbeat/processors/actions/drop_fields_test.go`, otherwise the CI (`libbeat-checks`) would not pass. It runs `goimport` and checks that there is no diff afterwards.

You can also run this step locally with `make check`.

**Comment by gonzaloe** [comment]
@rdner sorry for the inconvenience, I had first the problem with elastic-agent-libs and ran the update, but then with the new config package, but forgot to run make update now 😅  . I believe things should be in order

**Review by rdner — APPROVED** [review]
I see e2e tests passed here https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/main/906/pipeline but it's not reported to Github for some reason. I think it's okay to merge it now.

File: libbeat/processors/actions/drop_fields.go
Comment (rdner) [inline libbeat/processors/actions/drop_fields.go:77]: Sorry, just noticed this. `match.MustCompile` will panic if the value is invalid, that's not what we want, it should return an error like the one above on line 57.

**Review by rdner — CHANGES_REQUESTED** [review]

File: libbeat/processors/actions/drop_fields.go
Comment (gonzaloe) [reply libbeat/processors/actions/drop_fields.go]: 👍 done

File: libbeat/processors/actions/drop_fields.go
Comment (rdner) [inline libbeat/processors/actions/drop_fields.go:79]: I'd rather extend this error message to make it easier to debug. The test would need an update too.

```suggestion
				return nil, fmt.Errorf("wrong configuration in drop_fields[%d]=%s. %w", i, field, err)
```

Comment (gonzaloe) [reply libbeat/processors/actions/drop_fields.go]: oki sure

**Comment by gonzaloe** [comment]
@rdner good morning, E2E Tests are configured to run in every branch once the previous steps are successful? or only run in main/version branches?

**Comment by rdner** [comment]
@gonzaloe the E2E tests passed (https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/main/908/pipeline) I think something has been changed and they are not reported to Github correctly anymore. I'm going to approve the PR, we can merge it now.

**Review by rdner — APPROVED** [review]
LGTM,

@gonzaloe please open another PR with the documentation change.

**Pull request merged**
